### PR TITLE
Revamp plugin section

### PR DESCRIPTION
**Pull request by Issif**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind design

> /kind user-interface
/kind content

> /kind translation

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area blog

/area documentation

> /area videos

**What this PR does / why we need it**:

Revamp the plugins section

image

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: content/en/docs/plugins/developers-guide/how-to-register.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+| `docker.status`          | string | Status of the event            |
+| `docker.id`              | string | ID of the event                |
+| `docker.from`            | string | From of the event (deprecated) |
+| `docker.type`            | string | Type of the event              |
+| `docker.action`          | string | Action of the event            |
+| `docker.stack.namespace` | string | Stack Namespace                |
+
+# Propose your Plugin
+
+Once you're ready, you can submit your plugin to registration:
```
Comment (vjjmiras) [inline content/en/docs/plugins/developers-guide/how-to-register.md:82]:
```suggestion
Once you're ready, submit your plugin to registration:
```
```suggestion
Once you're ready, you can submit your plugin to registration:
```

File: content/en/docs/reference/plugins/plugin-api-reference.md [hunk 2]
```diff
@@ -798,6 +799,6 @@ State access is not thread-safe. Operations related to either discovery, reading
...
+There may be use cases in which state updates are the result of some asynchrnous job and computation. For example, the Falcosecurity libraries implement the container metadata enrichment support by connecting to one or more container runtime sockets and fetching information from them asynchronously from a separate thread, outside of the main event processing loop. In those cases, state updates must still happen synchronously. The strategy provided by the plugin framework is the [async events capability](/docs/reference/plugins/plugin-api-reference/#async-events-capability-api). With that, plugins are provided with a thread-safe callback that they can use to inject asynchronous events in the currently-open event stream, and the framework will guarantee those events to be later received by functions such as `parse_event` or `extract_fields` just like any other events. This massaging-like communication mode is the only way by which plugins are allowed to safely use asynchronously-obtained information to perform state updates and field extraction.
```
Comment (vjjmiras) [inline content/en/docs/reference/plugins/plugin-api-reference.md:802]:
```suggestion
There may be use cases when the state update results of some asynchronous job and computation. For example, the Falcosecurity libraries implement the container metadata enrichment support by connecting to one or more container runtime sockets and fetching information asynchronously using a separate thread of the main event processing loop. In those cases, state updates must still happen synchronously. The [async events capability](/docs/reference/plugins/plugin-api-reference/#async-events-capability-api) is the strategy provided by the plugin framework. With that, plugins are provided with a thread-safe callback that they can use to inject asynchronous events in the currently-open event stream, and the framework will guarantee those events to be later received by functions such as `parse_event` or `extract_fields` just like any other events. Plugins can only safely utilize asynchronously-obtained information for state updates and field extraction through this messaging-like communication mode.
```

File: content/en/docs/reference/plugins/plugin-api-reference.md [hunk 3]
```diff
@@ -798,6 +799,6 @@ State access is not thread-safe. Operations related to either discovery, reading
...
 
 An additional effect of injecting asynchronous events in an event stream is that they can so be recorder in SCAP capture files, and thus being reproducible when reading events later from that capture files. By having the asynchronous information recorded in the event stream, the event parsing and field extraction plugin functions will be able to work just like in live capture mode by also making all the state transitions reproducible and deterministic.
```
Comment (vjjmiras) [inline content/en/docs/reference/plugins/plugin-api-reference.md:804]:
```suggestion
An additional effect of injecting asynchronous events in an event stream is that they can so be recorded in SCAP capture files, thus being reproducible when reading events later from that capture files. By having the asynchronous information recorded in the event stream, the event parsing and field extraction plugin functions will be able to work just like in live capture mode by also making all the state transitions reproducible and deterministic.
```

File: content/en/docs/reference/plugins/plugin-api-reference.md [hunk 4]
```diff
@@ -2,9 +2,10 @@
 title: Falco Plugins API Reference
 linktitle: API Reference
 description: Learn how the Plugins API works
-weight: 60
+weight: 20
 aliases:
     - /docs/plugins/plugin_api_reference/
+    - /docs/reference/plugins/plugin_api_reference/
```
Comment (vjjmiras) [inline content/en/docs/reference/plugins/plugin-api-reference.md:8]: There is no need to add the actual location as an alias.

File: content/en/docs/reference/plugins/go-sdk-walkthrough.md [hunk 5]
```diff
@@ -2,9 +2,10 @@
 title: Falco Plugins Go SDK Walkthrough
 linktitle: Go SDK Walkthrough
 description: High-level documentation of the Go SDK
-weight: 50
+weight: 30
 aliases:
     - /docs/plugins/go_sdk_walkthrough/
+    - /docs/reference/plugins/go_sdk_walkthrough/
```
Comment (vjjmiras) [inline content/en/docs/reference/plugins/go-sdk-walkthrough.md:8]: There is no need to add the actual location as an alias.

**Review by leogr — APPROVED** [review]

**Pull request merged**